app/Http/Controllers/GroupsController.php

```diff
@@ -283,6 +283,37 @@ public function detach(Request $request)
         return $this->created('User has been removed from group');
     }
 
+    /**
+     * Promotes a user to leader of a group.
+     *
+     * @param integer
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function promote($id, Request $request)
```

Comment:
By naming the variable `$group` which is the same name in your web.php route `Route::put('/web/group/{group}/promote', 'GroupsController@promote');` (See `{group}`), Laravel will automatically run `Group::findOrFail($id)` and give inject it into the controller as a paramater. Your code is not wrong, this is just useful to know.

```suggestion
    public function promote(Group $group, Request $request)
```